src/Http/Senders/GuzzleSender.php

```diff
@@ -170,9 +170,10 @@ function (TransferException $guzzleException) use ($pendingRequest, $psrRequest)
 
                     $response = $this->createResponse($guzzleResponse, $pendingRequest, $psrRequest, $guzzleException);
 
-                    // Throw the exception our way
+                    // Throw the exception our way if there is an exception.
+                    // Otherwise we'll return the response.
 
-                    throw $response->toException();
+                    return ($exception = $response->toException()) ? throw $exception : $response;
```

Comment:
I'm a little confused by this one - do you mind explaining a little please?

Reply:
The `$response->toException()` method can return null if the `shouldThrowRequestException()` method returns null. In that case, we can not throw an exception and we have to return the response.

Reply:
Ah gotcha